### ISSUE TITLE
Review error / status messages

### ISSUE THREAD

**Issue by DeeEmm** [DeeEmm/DIY-Flow-Bench#77]
Need to tidy up error and status message handling

**Comment by DeeEmm** [comment]
Removed status message reporting from API - API now only returns status via Serial link. It made no sense to have API triggering messages on the Web-UI.

Added a 'boot message'. This is the default value used for the System Status Message and will be visible in the WebUI on initial power-up. This both initialises the Message field and gives users something to tweak. BOOT_MESSAGE is defined within the configuration header file (configuration.h)

**Comment by DeeEmm** [comment]
Need to detect size of mafData array and then do some maths to interpolate data values. 

For a 12 bit array we can just use the raw value, but for a 10 bit array our 12 bit reference will not work, so we need to work out which two values fall either side of our reference value and then do some interpolation to calculate the actual value. We could of course limit the reference value to 10 bits (probably a good idea initially) but then we lose that extra resolution.

We are already doing a similar thing with the key>value style mafData array as this is not a fixed length array.

There we are using the following interpolation formula:

```
Y=Y0+(((X-X0)(Y1-Y0))/(X1-X0)) 

where Y = flow and X = Volts
```

We might even be able to recycle this code and move it into a function

**Comment by DeeEmm** [comment]
Need to determine what message are displayed to the user via the WebUI.

- [ ] Error messages
- [ ] Status messages
- [ ] Confirmation messages

Need to determine what clears down error messages

- [ ] User confirmation
- [ ] Error remedied
- [ ] Continued operation (error ignored but no longer active)

**Comment by DeeEmm** [comment]
API Status messages were being reported to the GUI status field. This was also related to #97 where the status message was incorrectly being sent by debug.print method

Moved API status messages out of global scope and changed to serial.print method.

**Comment by DeeEmm** [comment]
Message: 'Flow limit reached'

Description: 'Test object flow exceeds Vac source capability'

Need to display to user when measured flow approaches maximum flow capability of vac source.

Also need to include config setting to set threshold
Also need to be able to calibrate this value i.e. record flow with no test piece

**Comment by DeeEmm** [comment]
Some further thoughts on this...

Bench limits can be determined by testing the reference depression. If depression cannot be maintained when increasing flow requirements the bench has reached it's limit. This can be used to determine the flow limit for the bench.

There's essentially two ways I can see this working

1. A range of orifices can be measured to determine what the maximum achievable flow is for a given reference pressure. This flow value can then be 'hard coded' by manually entering it as a configuration parameter.
2. The depression value can be compared to vacuum control 'on-the-fly'. Limit is reached when vacuum control is at or approaching 100% modulation, but reference pressure cannot be maintained

Basically if the reference depression cannot be maintained for a given flow rate, then test piece is outflowing the vac source.

There are two ways to set this up:

1. Manually enter the calculated / measured limit and then compare flow rate to this limit. i.e. if flow_rate >= flow_rate_limit
2. Test limit 'on-the-fly' by comparing vac_speed_ref to ref_pressure. This does not require any specific limit to be measured or set and works for all depressions / flow rates

Option 2 is more flexible but requires integrated vac speed control. Option 1 will work for any setup but requires that the user sets up the initial flow limit.
